tests/test_leftovers.py

```diff
@@ -0,0 +1,294 @@
+    fake_did = "testprefix:d96bab16-c4e1-44ac-923a-04328b6fe78f"
+    with pytest.raises(IndexNoRecordFound):
+        get_record(fake_did)
+
+
+def test_get_dos_record(client, user):
+    data = get_doc(has_urls_metadata=True, has_metadata=True, has_baseid=True)
+
+    res_1 = client.post("/index/", json=data, headers=user)
+    assert res_1.status_code == 200
+
+    # test exception raised at nonexistent
+    fake_did = "16e51a9a-af23-45a3-b4b2-94ae2b143feb"
+    with pytest.raises(IndexNoRecordFound):
+        get_dos_record(fake_did)
+
+
+def test_alias_list_by_hash(client, user):
+    data = {
+        "size": 123,
+        "hashes": {"md5": "8b9942cf415384b27cadf1f4d2d682e5"},
+        "release": "private",
+        "keeper_authority": "CRI",
+        "host_authorities": ["PDC"],
+    }
+    ark = "ark:/31807/TEST-abc"
+    res = client.put("/alias/" + ark, json=data, headers=user)
+    assert res.status_code == 200
+    rec = res.json
+    assert rec["name"] == ark
+    # assert there is only one entry in the alias index
+    assert len(client.get("/alias/").json["aliases"]) == 1
+    res = client.get("/alias/?hash=" + "md5:8b9942cf415384b27cadf1f4d2d682e5")
+    rec = res.json
+    # assert that the returned alias by hash is the same as the posted
+    assert rec["aliases"][0] == ark
+    assert rec["hashes"] == data["hashes"]
+
+
+def test_alias_list_by_size(client, user):
+    data = {
+        "size": 123,
+        "hashes": {"md5": "8b9942cf415384b27cadf1f4d2d682e5"},
+        "release": "private",
+        "keeper_authority": "CRI",
+        "host_authorities": ["PDC"],
+    }
+    ark = "ark:/31807/TEST-abc"
+    res = client.put("/alias/" + ark, json=data, headers=user)
+    assert res.status_code == 200
+    rec = res.json
+    assert rec["name"] == ark
+    # assert there is only one entry in the alias index
+    assert len(client.get("/alias/").json["aliases"]) == 1
+    # assert that the returned alias by size is the same as the posted
+    res = client.get("/alias/?size=" + str(data["size"]))
+    rec = res.json
+    assert rec["aliases"][0] == ark
+
+
+def test_index_stats(client, user):
+    # populate the index with three different size records
+    data1 = get_doc()
+    res = client.post("/index/", json=data1, headers=user)
+    assert res.status_code == 200
+    data2 = get_doc()
+    data2["size"] = 77
+    res = client.post("/index/", json=data2, headers=user)
+    assert res.status_code == 200
+    data3 = get_doc()
+    data3["size"] = 300
+    res = client.post("/index/", json=data3, headers=user)
+    assert res.status_code == 200
+    data_size = data1["size"] + data2["size"] + data3["size"]
+    index_stats = client.get("/_stats/").json
+    file_number = index_stats["fileCount"]
+    index_file_size = index_stats["totalFileSize"]
+    # test that the stat file number and size is consistent with post
+    assert file_number == 3
+    assert index_file_size == data_size
```

Comment:
cool test!!

you could make the assertions more compact by using directly `assert index_stats["fileCount"] == 3` and `assert index_stats["totalFileSize"] == data_size`